### ISSUE TITLE
Could not locate the bindings file, with `gl`

### ISSUE THREAD

**Issue by bthj** [oven-sh/bun#5357]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

When using a dependency of [gl](https://www.npmjs.com/package/gl) (via [GPU.js](https://github.com/gpujs/gpu.js)), the error "Could not locate the bindings file" is printed.

The following steps produce this bug:
```
mkdir bun-test
cd bun-test
bun init
```
- specify `index.js` as the entry point and add the following to it:
```
const { GPU } = require('gpu.js');
const gpu = new GPU();
```
- then run:
```
bun add gpu.js
bun run index.js
```

### What is the expected behavior?

There should be no error.

### What do you see instead?

```
% bun run index.js
121 |         throw e;
122 |       }
123 |     }
124 |   }
125 | 
126 |   err = new Error(
            ^
error: Could not locate the bindings file. Tried:
 → /Users/bthj/bun-test/node_modules/gl/build/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/build/Debug/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/build/Release/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/out/Debug/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/Debug/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/out/Release/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/Release/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/build/default/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/compiled/18.15.0/darwin/arm64/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/addon-build/release/install-root/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/addon-build/debug/install-root/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/addon-build/default/install-root/webgl.node
 → /Users/bthj/bun-test/node_modules/gl/lib/binding/node-v108-darwin-arm64/webgl.node
      at bindings (/Users/bthj/bun-test/node_modules/bindings/bindings.js:126:8)
      at /Users/bthj/bun-test/node_modules/gl/src/javascript/native-gl.js:1:6
      at globalThis (/Users/bthj/bun-test/node_modules/gl/src/javascript/native-gl.js:13:64)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gl/src/javascript/webgl-rendering-context.js:4:41
      at globalThis (/Users/bthj/bun-test/node_modules/gl/src/javascript/webgl-rendering-context.js:3864:54)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gl/src/javascript/node-index.js:3:31
      at globalThis (/Users/bthj/bun-test/node_modules/gl/src/javascript/node-index.js:129:17)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gl/index.js:4:9
      at globalThis (/Users/bthj/bun-test/node_modules/gl/index.js:4:9)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gpu.js/src/backend/headless-gl/kernel.js:1:6
      at globalThis (/Users/bthj/bun-test/node_modules/gpu.js/src/backend/headless-gl/kernel.js:145:0)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gpu.js/src/gpu.js:5:8
      at globalThis (/Users/bthj/bun-test/node_modules/gpu.js/src/gpu.js:605:0)
      at require (:1:20)
      at /Users/bthj/bun-test/node_modules/gpu.js/src/index.js:1:8
      at globalThis (/Users/bthj/bun-test/node_modules/gpu.js/src/index.js:57:0)
      at require (:1:20)
      at /Users/bthj/bun-test/index.js:1:8
      at globalThis (/Users/bthj/bun-test/index.js:2:16)
```

**Comment by controversial** [comment]
The `gl` module has scripts that need to be run on install in order to compile native code. Bun does not run arbitrary package scripts by default, for security purposes.

You need to add `gl` to `trustedDependencies` in your `package.json`:

```json
{
  "dependencies": {
    "gpu.js": "^2.16.0"
  },
  "trustedDependencies": ["gl"]
}
```

**Comment by bthj** [comment]
Thanks @controversial. Running a fresh `bun install` with those settings in place leads to a new error:
```
% bun index.js 
dyld[33489]: missing symbol called
zsh: killed     bun index.js
```
Probably this has something to do with running on Apple silicon (M1 Pro) ? 

For Node.js I can see [suggestions on reinstalling within a Rosetta2 shell](https://gist.github.com/adrienjoly/e29a6e73fb7e701eefd80ff9bde9abeb?permalink_comment_id=4105601) - is that also relevant for bun? Or are there some other settings I could try out?

**Comment by controversial** [comment]
Does it work in Node?

**Comment by bthj** [comment]
Yes, this works in Node.

**Comment by controversial** [comment]
If it works in Node, it’s likely a bug (or missing feature) in Bun

**Comment by matsumoto0912** [comment]
Same error when using better-sqlite3

https://github.com/oven-sh/bun/issues/7448

**Comment by Jarred-Sumner** [comment]
Fixed by @dylan-conway and @paperdave  in #7132

**Comment by hanselke** [comment]
getting error for gpu.js
bun: symbol lookup error: /data/projectName/node_modules/gl/build/Release/webgl.node: undefined symbol: node_module_register

using today's build, with "trustedDependencies": ["gl"] and a fresh bun install.
